### ISSUE TITLE
Experimental trait cannot be instantiated anonymously

### ISSUE THREAD

**Issue by prolativ** [lampepfl/dotty#13091]
## Compiler version

3.0.3-RC1-bin-20210715-7899462-NIGHTLY

## Minimized code

```Scala
import annotation.experimental
@experimental trait Foo
val foo = new (Foo @experimental) {}
```

## Output

```scala
-- Error:
3 |val foo = new (Foo @experimental) {}
  |          ^
  |    extension of experimental trait Foo must have @experimental annotation
```

## Expectation

It should be possible to create an instance of a trait marked as experimental without having to create a helper class like

```
@experimental class Bar extends Foo
val bar = new Bar
```

The problem seems to also affect classes with a refinement.

**Comment by odersky** [comment]
I wonder why have that restriction in the first place? cc @nicolasstucki @sjrd

**Comment by sjrd** [comment]
Doing *anything* with something experimental is experimental. But since there is already an
```scala
import annotation.experimental
```
it seems that it should be allowed.

Then again, it could also be because of the transitivity argument: if you expose a class that extends something experimental, you are in fact exposing that experimental thing as part of your hierarchy, even though you're not experimental. But I don't think we've had this discussion before, so the current restriction is probably not intentional, at least.

**Comment by odersky** [comment]
But I don't see how extends is different from uses. Classes using an experimental class don't need to be marked as experimental.

**Comment by sjrd** [comment]
When you use something, it stays private to your implementation, usually it's fine. If you inherit from something, it always leaks. (It's already a problem with the fact that Scala lets you extend a class or trait with a lesser visibility than yours, which causes actual problems by leaking package-private stuff.)

But like I said, we can argue that, but it wasn't part of the discussions/designs, so the restriction should probably be lifted. (Or, if we decide so, consistently applied to every use that leaks in the public API.)

**Comment by odersky** [comment]
We have to go over the books here. I almost fell from my chair when I looked at all the anti-experimental checks RefChecks. What are we trying to achieve here anyway?

**Comment by odersky** [comment]
I wanted to do a quick fix in #13093 but now realize it will be a huge mess to even update all the tests. So  I'll refrain from that. But it should still be done and cut down to the bare minimum. And be done quickly. Experimental is just out, and will be a major hassle for many people. We should err on the side of caution. It seems what we did instead is go completely over the top.

**Comment by nicolasstucki** [comment]
We do not require the annotation on anonymous classes. This one slipped through the checks.
